tests/test_data_loader.py

```diff
@@ -39,4 +39,4 @@ def test_preprocess_data(mocker, mock_df: pd.DataFrame):
         index=[0, 1, 2],
     )
     expected_df.index.name = "id"
-    assert_frame_equal(data_loader.party_data, expected_df, check_less_precise=3)
+    assert_frame_equal(data_loader.party_data, expected_df, rtol=3)
```

Comment:
The test failed for me:

>       assert_frame_equal(data_loader.party_data, expected_df, rtol=3)
E       AssertionError: DataFrame are different
E       
E       DataFrame shape mismatch
E       [left]:  (4, 6)
E       [right]: (3, 3)